### ISSUE TITLE
Standardized Experiment Runner for V-A Regression to faciliate documentation and comparison

### ISSUE THREAD

**Issue by DesmondChoy** [keimiii/emo-rec#10]
# Standardized Experiment Runner for V-A Regression

This notebook provides a standardized environment for conducting model training experiments, ablation studies, and comparisons for our academic project. 

The primary goal is to ensure that every experiment is reproducible and generates all the necessary artifacts and metrics required for our analysis. By running a model configuration through this notebook, you will automatically populate all the criteria outlined in our project's evaluation checklist.

# File Location

notebooks/scene/generic_model.ipynb

# Checklist

***

* **Dataset**
    * The dataset is loaded and prepared in the **"Load FindingEmo dataset"** section.

* **Train/Validation/Test Split**
    * The final split sizes are printed after the dataframes are loaded and cleaned in the **"Load FindingEmo dataset"** section.

* **Architecture Backbone**
    * Defined by the `BACKBONE_SRC` and `MODEL_NAME` variables in the **"Config"** section.

* **Architecture Head**
    * The head is defined in the `BackboneRegressor` class. Its complexity can be modified via the `hidden` parameter during instantiation.

* **Base Width & Height**
    * Set by the `BASE_W` and `BASE_H` variables in **"Config"**. The final computed input size is verified before training.

* **Augmentation**
    * The augmentation pipeline is defined using `aug_transforms` within the `df_to_dls` function in the **"DataBlock & DataLoaders"** section.

* **Parameters (Epochs, LR, Loss)**
    * Key parameters like `EPOCHS`, `loss_func` (MSE), and the `opt_func` (Adam) are set in the **"Config"** and **"Find Optimal LR"** sections.

* **Learning/Loss Curve**
    * The training and validation loss curves are plotted by the cell under the **"Plot training/validation loss curves"** heading. Now showing Epochs too.

<img width="576" height="455" alt="Image" src="https://github.com/user-attachments/assets/5ea1bffd-260e-4cde-9f56-6f0822e2a5f6" />

* **Early stopping implemented**

<img width="725" height="471" alt="Image" src="https://github.com/user-attachments/assets/4793f648-68c2-4f5c-be30-0fe7a5723377" />

* **Training & Test Evaluation**
    * All final metrics (Valence/Arousal MAE, Spearman's R, Test Loss (MSE), etc.) are calculated and displayed in a dictionary in the **"Evaluate on Test Set"** section.

* **Saved Weights Path**
    * The trained head weights are saved, and the path is printed in the final **"Save head weights"** section.

**Comment by DesmondChoy** [comment]
@keimiii @kevinmanuellee @jrlyy I ran this on resnet50 & DINOv3 and it works. Let me know if you encounter any bugs!
